src/client/options.rs

```diff
@@ -1721,6 +1750,26 @@ impl ConnectionString {
             ConnectionStringParts::default()
         };
 
+        if let Some(srv_max_hosts) = conn_str.srv_max_hosts {
```

Comment:
Similarly, the spec says that the validation here _has_ to happen both for the initial URI parse and after SRV resolution, so that gets duplicated too :/